table/neuron/table.cpp

```diff
@@ -0,0 +1,409 @@
+    void check_sigmoid1_tbl(_nrn_mechanism_cache_range* _ml, tbl_Instance& inst, size_t id, Datum* _ppvar, Datum* _thread, NrnThread* _nt);
+    static void _check_table_thread(Memb_list* _ml, size_t id, Datum* _ppvar, Datum* _thread, NrnThread* _nt, int _type, _nrn_model_sorted_token const& _sorted_token)
+{
+        _nrn_mechanism_cache_range _lmr{_sorted_token, *_nt, *_ml, _type};
+        auto inst = make_instance_tbl(_lmr);
+        check_sigmoid1_tbl(&_lmr, inst, id, _ppvar, _thread, _nt);
+    }
+
+
+    /** connect global (scalar) variables to hoc -- */
+    static DoubScal hoc_scalar_double[] = {
+        {"k_tbl", &tbl_global.k},
+        {"d_tbl", &tbl_global.d},
+        {"usetable_tbl", &tbl_global.usetable},
+        {nullptr, nullptr}
+    };
+
+
+    /** connect global (array) variables to hoc -- */
+    static DoubVec hoc_vector_double[] = {
+        {nullptr, nullptr, 0}
+    };
+
+
+    /* declaration of user functions */
+    static void _hoc_sigmoid1(void);
+    static double _npy_sigmoid1(Prop*);
+
+
+    /* connect user functions to hoc names */
+    static VoidFunc hoc_intfunc[] = {
+        {"setdata_tbl", _hoc_setdata},
+        {"sigmoid1_tbl", _hoc_sigmoid1},
+        {0, 0}
+    };
+    static NPyDirectMechFunc npy_direct_func_proc[] = {
+        {"sigmoid1", _npy_sigmoid1},
+    };
+    static void _hoc_sigmoid1(void) {
+        double _r{};
+        Datum* _ppvar;
+        Datum* _thread;
+        NrnThread* _nt;
+        Prop* _local_prop = _prop_id ? _extcall_prop : nullptr;
+        _nrn_mechanism_cache_instance _ml_real{_local_prop};
+        auto* const _ml = &_ml_real;
+        size_t const id{};
+        _ppvar = _local_prop ? _nrn_mechanism_access_dparam(_local_prop) : nullptr;
+        _thread = _extcall_thread.data();
+        _nt = nrn_threads;
+        auto inst = make_instance_tbl(_ml_real);
+        check_sigmoid1_tbl(_ml, inst, id, _ppvar, _thread, _nt);
+        _r = 1.;
+        sigmoid1_tbl(_ml, inst, id, _ppvar, _thread, _nt, *getarg(1));
+        hoc_retpushx(_r);
+    }
+    static double _npy_sigmoid1(Prop* _prop) {
+        double _r{};
+        Datum* _ppvar;
+        Datum* _thread;
+        NrnThread* _nt;
+        _nrn_mechanism_cache_instance _ml_real{_prop};
+        auto* const _ml = &_ml_real;
+        size_t const id{};
+        _ppvar = _nrn_mechanism_access_dparam(_prop);
+        _thread = _extcall_thread.data();
+        _nt = nrn_threads;
+        auto inst = make_instance_tbl(_ml_real);
+        check_sigmoid1_tbl(_ml, inst, id, _ppvar, _thread, _nt);
+        _r = 1.;
+        sigmoid1_tbl(_ml, inst, id, _ppvar, _thread, _nt, *getarg(1));
+        return(_r);
+    }
+
+
+    inline int f_sigmoid1_tbl(_nrn_mechanism_cache_range* _ml, tbl_Instance& inst, size_t id, Datum* _ppvar, Datum* _thread, NrnThread* _nt, double v) {
+        int ret_f_sigmoid1 = 0;
+        inst.sig[id] = 1.0 / (1.0 + exp(inst.global->k * (v - inst.global->d)));
+        return ret_f_sigmoid1;
```

Comment:
When would this function return a non-zero exit code? If never, we could use the return value for `sig(v)`. If there's multiple return values, output arguments, might be an option.

Reply:
Quoting from [the docs](https://nrn.readthedocs.io/en/latest/python/modelspec/programmatic/mechanisms/nmodl.html#procedure):
> If a PROCEDURE is solved by a SOLVE statement it is necessary for the procedure to return an error code, (0 denotes success). 

so I guess it's possible there are `PROCEDURE`s which actually do return non-zero exit codes in the wild (haven't encountered any though).